src/d3a/models/state.py

```diff
@@ -175,6 +174,7 @@ def clamp_energy_to_sell_kWh(self, market_slot_time_list):
                                                             energy / len(market_slot_time_list),
                                                             self.max_offer_energy_kWh(time_slot),
                                                             self._battery_energy_per_slot))
+            self.energy_to_sell_dict[time_slot] = storage_dict[time_slot]
```

Comment:
I dont see another place where this dict is used. Is there a plan to use it a future ticket?

Reply:
Its needed to be sure if external agent is not over-selling. Also added an assert in recent [commit](https://github.com/gridsingularity/d3a/pull/639/files#diff-1d553c99bd95475906818e1177a93f29R95)